src/MongoDB.Driver.Core/Core/WireProtocol/Messages/Encoders/BinaryEncoders/CommandMessageBinaryEncoder.cs

```diff
@@ -92,6 +92,8 @@ public void WriteMessage(CommandMessage message)
             stream.WriteInt32((int)CreateFlags(message));
             WriteSections(writer, message.Sections, messageStartPosition);
             stream.BackpatchSize(messageStartPosition);
+
+            message.PostWriteAction?.Invoke(new PostProcessor(message, stream, messageStartPosition));
```

Comment:
This is the call to the hook that can change the write concern from { w : 0 } to { w : 1 } if necessary.